services/controllerhost/controllerhost.go

```diff
@@ -173,7 +174,8 @@ func NewController(cfg configure.CommonAppConfig, sVice *common.Service, metadat
 
 	context.dstLock = lockMgr
 	context.m3Client = metrics.NewClient(instance.Service.GetMetricsReporter(), metrics.Controller)
-	context.mm = common.NewMetadataMgr(metadataClient, context.m3Client, context.log)
+	mClient := metadata.NewMetadataMetricsMgr(metadataClient, context.m3Client, context.log)
```

Comment:
you should probably update the global (Mcp.mClient) to use the instance that emits metrics .. so that other sub-components (within controllerhost) that calls into metadata would automatically emit metrics. i see that retention manager instantiate another instance of metadata-metrics within its code -- we should probably remove that.